pkg/kube/activity.go

```diff
@@ -227,6 +247,145 @@ func (k *PipelineActivityKey) GetOrCreate(jxClient versioned.Interface, ns strin
 	}
 }
 
+func (k *PipelineActivityKey) reconcileBatchBuildIndividualPR(activitiesClient typev1.PipelineActivityInterface, currentActivity *v1.PipelineActivity) error {
+	log.Info("Checking if batch build reconciling is needed")
+	//Create a selector for other runs of this PR with the same last commit SHA
+	labels := currentActivity.Labels
+	selector := fmt.Sprintf("lastCommitSha in (%s), branch in (%s), sourcerepository in (%s)",
+		labels[v1.LabelLastCommitSha], labels[v1.LabelBranch], labels[v1.LabelSourceRepository])
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: selector,
+	}
+
+	log.Debugf("looking for PipelineActivities with selector %s", selector)
+	activities, err := activitiesClient.List(listOptions)
+	if err != nil {
+		return errors.Wrap(err, "there was a problem listing all activities to reconcile a batch build")
+	}
+
+	if len(activities.Items) == 0 {
+		log.Infof("No past executions with the same lastCommitSha found - reconciliation not needed")
+		return nil
+	}
+
+	currentBuildNumber, err := strconv.Atoi(k.Build)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing the current build number for PipelineActivity %s", currentActivity.Name)
+	}
+
+	//It only makes sense to look for the build before this one
+	currentBuildNumber = currentBuildNumber - 1
+	for _, v := range activities.Items {
```

Comment:
maybe this could be extracted into a method which describes what's going on here.

Reply:
I think the resulting method would be messy due to all the parameters that it would need. I'd always to keep params lower than 3. 

I will however word it better so it's more understandable.

Reply:
ok

> I think the resulting method would be messy due to all the parameters that it would need

which might point to a deeper lying issue. But anyways